app/code/Magento/Deploy/Model/Deployer.php

```diff
@@ -105,20 +152,59 @@ function (AlternativeSourceInterface $alternative) {
             $alternativeSources
         );
         $this->alternativeSources = $alternativeSources;
+
+    }
+
+    /**
+     * Check if skip flag is affecting file by extension
+     *
+     * @param string $filePath
+     * @return boolean
+     * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+     * @SuppressWarnings(PHPMD.NPathComplexity)
+     */
+    private function checkSkip($filePath) {
+        $path = $filePath;
+        $ext = pathinfo($path, PATHINFO_EXTENSION);
+
+        $check = ($this->isJavaScript || $this->isCss || $this->isLess || $this->isHtml || $this->isImages || $this->isFonts || $this->isMisc);
+
+        if ($check && $filePath != '.') {
+            if ($this->isJavaScript && in_array($ext, array('js', 'map'))) {
+                return true;
+            } else if ($this->isCss && $ext == 'css') {
+                return true;
+            } else if ($this->isLess && $ext == 'less') {
+                return true;
+            } else  if ($this->isHtml && in_array($ext, array('html', 'htm'))) {
+                return true;
+            } else  if ($this->isImages && in_array($ext, array('jpg', 'jpeg', 'gif', 'png', 'ico', 'svg'))) {
+                return true;
+            } else  if ($this->isFonts && in_array($ext, array('eot', 'svg', 'ttf', 'woff', 'woff2'))) {
+                return true;
+            } else  if ($this->isMisc && in_array($ext, array('md', 'txt', 'jbf', 'csv', 'json', 'txt', 'htc', 'swf', 'LICENSE', ''))) {
```

Comment:
I would rather you put those file extension strings somewhere, if not configurable, at least more visible, like some private properties.